person.rb

```diff
@@ -1,12 +1,61 @@
-class Person
+class Nameable
+  def initialize(name)
+    @nameable = name
+  end
+
+  def correct_name
+    raise NotImplementedError, "#{self.class} has not implemented methods '#{__method__}'"
+  end
+end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    @nameable = nameable
+    super
+  end
+
+  def correct_name
+    @nameable
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def initialize(nameable)
+    @nameable = nameable
+    super
+  end
+
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def initialize(nameable)
+    @nameable = nameable
+    super
+  end
+
+  def correct_name
+    return @nameable.correct_name.slice(0, 10) if @nameable.correct_name.length > 10
```

Comment:
- [ ] this method should return the first 10 characters of the `#nameable` variable however your code is returning the first 11.